Sources/HttpPipeline/Middleware.swift

```diff
@@ -86,24 +85,26 @@ public func send(_ data: Data?) -> Middleware<BodyOpen, BodyOpen, Data?, Data?>
   }
 }
 
-public func respond<A>(text: String) -> Middleware<HeadersOpen, ResponseEnded, A, Data?> {
+public func respond<A>(text: String) -> Middleware<HeadersOpen, ResponseEnded, A, Data> {
   return respond(body: text, contentType: .plain)
 }
 
-public func respond<A>(html: String) -> Middleware<HeadersOpen, ResponseEnded, A, Data?> {
+public func respond<A>(html: String) -> Middleware<HeadersOpen, ResponseEnded, A, Data> {
   return respond(body: html, contentType: .html)
 }
 
-public func respond<A>(json: String) -> Middleware<HeadersOpen, ResponseEnded, A, Data?> {
+public func respond<A>(json: String) -> Middleware<HeadersOpen, ResponseEnded, A, Data> {
   return respond(body: json, contentType: .json)
 }
 
 public func respond<A>(body: String, contentType: MediaType)
-  -> Middleware<HeadersOpen, ResponseEnded, A, Data?> {
-    let data = body.data(using: .utf8)
+  -> Middleware<HeadersOpen, ResponseEnded, A, Data> {
+
+    let data = Data(body.utf8)
+
     return map(const(data)) >>> pure
       >-> writeHeader(.contentType(contentType))
-      >-> writeHeader(.contentLength(data?.count ?? 0))
+      >-> writeHeader(.contentLength(data.count))
```

Comment:
is this equivalent to `body.utf8.count`?

Reply:
never mind I didnt see that `data` was being used elsewhere. it thought we could avoid the `Data` creation...